src/dialogs/copyPasteOwnership.tsx

```diff
@@ -0,0 +1,185 @@
+
+import { useFilesContext } from '../app.tsx';
+import type { ClipboardInfo } from '../app.tsx';
+import { get_owner_candidates } from '../ownership.tsx';
+
+const _ = cockpit.gettext;
+
+async function pasteAsOwner(clipboard: ClipboardInfo, dstPath: string, owner: string,
+    addAlert: (title: string, variant: AlertVariant, key: string, detail?: string) => void) {
+    try {
+        await cockpit.spawn([
+            "cp",
+            "--recursive",
+            ...clipboard.files.map(file => clipboard.path + "/" + file.name),
+            dstPath
+        ], { superuser: "try" });
+
+        await cockpit.spawn([
+            "chown",
+            "--recursive",
+            owner,
+            ...clipboard.files.map(file => dstPath + "/" + file.name),
+        ], { superuser: "try" });
+    } catch (err) {
+        const e = err as cockpit.BasicError;
+        addAlert(e.message, AlertVariant.danger, `${new Date().getTime()}`);
+
+        // cleanup potentially copied files in case of "chown" fail
+        try {
+            await cockpit.spawn([
+                "rm",
+                "-rf",
+                ...clipboard.files.map(file => dstPath + "/" + file.name)
+            ], { superuser: "try" });
+        } catch (ex) {
+            console.warn(`Failed to clean up copied files in ${dstPath}`, ex);
+        }
+    }
+}
+
+function makeCandidatesMap(currentUser: cockpit.UserInfo, cwdInfo: FileInfo, clipboard: ClipboardInfo) {
+    const map: Map<string, string> = new Map();
+    const candidates = get_owner_candidates(currentUser, cwdInfo);
+
+    // also add current ownership if it is same for all files (shallow check)
+    const firstFile = clipboard.files[0];
+    const firstOwnerStr = `${firstFile.user}:${firstFile.group}`;
+    if (clipboard.files.every(file => file.user === firstFile.user && file.group === firstFile.group)) {
+        candidates.add(firstOwnerStr);
+    }
+
+    candidates.forEach(owner => {
+        const split = owner.split(':');
+        let key = owner;
+        if (split.length === 2 && split[0] === split[1]) {
+            key = split[0];
+        }
+        if (owner === firstOwnerStr) {
+            key = `${key} ${_("(original owner)")}`;
+        }
+
+        map.set(key, owner);
+    });
+
+    return map;
+}
+
+const CopyPasteAsOwnerModal = ({
+    clipboard,
+    dialogResult,
+    path,
+} : {
+    clipboard: ClipboardInfo,
+    dialogResult: DialogResult<void>,
+    path: string,
+}) => {
+    const [currentUser, setCurrentUser] = useState<cockpit.UserInfo| undefined>();
+    const [selectedOwner, setSelectedOwner] = useState<string | undefined>();
+    const { cwdInfo, addAlert } = useFilesContext();
+
+    useEffect(() => {
+        cockpit.user().then(user => setCurrentUser(user));
+    }, []);
+
+    let candidatesMap: Map<string, string> = new Map();
+    if (superuser.allowed && currentUser && cwdInfo) {
+        candidatesMap = makeCandidatesMap(currentUser, cwdInfo, clipboard);
```

Comment:
This happens on every re-render, across the useInit/useEffect. Wouldn't it make more sense to move this into useEffect() as well, and initialize the map just once?